Improve Gammapy installation docs a bit

cdeil opened the pull request:
Here I improve the Gammapy installation docs a bit, using `python -m pip` everywhere and adding a link to http://jakevdp.github.io/blog/2017/12/05/installing-python-packages-from-jupyter/ as well as a few other small (hopefully) improvements.

I also removed the recommendation to use `linkchecker` when making a Gammapy - it doesn't work with Python 3, is unmaintained and it's forks or competing Python projects are all not great. It's also not really needed; the Sphinx `linkcheck` is already available for anyone working on the Sphinx docs and works well.

cc @adonath

adonath commented on docs/install/index.rst:
```diff
@@ -40,30 +40,33 @@ install it with e.g. ``apt-get`` or ``yum``.
 -------------------
 
-To install the development version of Gammapy::
+You can install the development version of Gammapy like this::
```
Maybe mention this is the recommended way for a conda environment as well?

cdeil replied:
@adonath - I think for the install and conda install we'll need follow-up PRs to improve what we have.
(note that there's still quite a bit of content in http://docs.gammapy.org/en/latest/notebooks/tutorial_setup.html)

cdeil commented:
I've continued with the docs a little bit here:

- Removed "news" from the docs front page. This will appear as a gammapy.org/news.html tomorrow.
- Moved more inline URLs to `references.txt` to make the RST text more maintainable, and changed a bunch of external links where `make linkcheck` showed that it's broken (very few) or permanently redirected (many).
